packages/db-ra-data-rest/src/index.js

```diff
@@ -0,0 +1,153 @@
+        acc[`where.${param}.eq`] = filters[param];
+
+        return acc;
+      }, {})
+    : {};
+
+export default (apiUrl, httpClient = fetchUtils.fetchJson) => ({
+  getList: (resource, params) => {
+    const { page, perPage } = params.pagination;
+    const { field, order } = params.sort;
+
+    const query = {
+      ...formatFilters(params.filter),
+      ...(order && { [`orderby.${field}`]: order.toLowerCase() }),
+      limit: perPage,
+      offset: (page - 1) * perPage,
+      totalCount: true,
+    };
+
+    const url = `${apiUrl}/${resource}?${stringify(query)}`;
+
+    return httpClient(url).then(({ headers, json }) => {
+      if (!headers.has("x-total-count")) {
+        throw new Error(
+          "The X-Total-Count header is missing in the HTTP Response. The jsonServer Data Provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?"
+        );
+      }
+      return {
+        data: json,
+        total: parseInt(headers.get("x-total-count").split("/").pop(), 10),
+      };
+    });
+  },
+
+  getOne: (resource, params) =>
+    httpClient(`${apiUrl}/${resource}/${params.id}`).then(({ json }) => ({
+      data: json,
+    })),
+
+  getMany: (resource, params) => {
+    const query = {
+      "where.id.in": params.ids.join(","),
+    };
+    const url = `${apiUrl}/${resource}?${stringify(query)}`;
+    return httpClient(url).then(({ json }) => ({ data: json }));
+  },
+
+  getManyReference: (resource, params) => {
+    const { page, perPage } = params.pagination;
+    const { field, order } = params.sort;
+
+    const query = {
+      ...formatFilters(params.filter),
+      [`where.${params.target}.eq`]: params.id,
+      [`orderby.${field}`]: order.toLowerCase(),
+      limit: perPage,
+      offset: (page - 1) * perPage,
+      totalCount: true,
+    };
+    const url = `${apiUrl}/${resource}?${stringify(query)}`;
+
+    return httpClient(url).then(({ headers, json }) => {
+      if (!headers.has("x-total-count")) {
+        throw new Error(
+          "The X-Total-Count header is missing in the HTTP Response. The jsonServer Data Provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?"
+        );
+      }
+      return {
+        data: json,
+        total: parseInt(headers.get("x-total-count").split("/").pop(), 10),
+      };
+    });
+  },
+
+  update: (resource, params) =>
+    httpClient(`${apiUrl}/${resource}/${params.id}`, {
+      method: "PUT",
+      body: JSON.stringify(params.data),
+    }).then(({ json }) => ({ data: json })),
+
+  // platformatic doesn't handle filters on UPDATE route, so we fallback to calling UPDATE n times instead
```

Comment:
Would you mind opening a separate issue about this? Looks like it would be useful.

Reply:
do you want me also to leave those comments?

Reply:
Leave the comments as they are but add a link to the issues. thx!